constructors.md

```diff
@@ -1 +1,112 @@
 # Constructors
+
+Assume that `DType` is the same as `gdf_dtype`, and that `device_buffer` is effectively a `rmm::device_vector` and `host_buffer` is effectively a `std::vector`. 
+
+Note: This design document purposefully ignores the issue of `allocators` and controlling how the device memory for the column is allocated. This will be added as the allocator design in fleshed out. For the timebeing, assume that an allocator handle can be passed into each of the constructors that controls how the device memory for the data and bitmask is allocated.
+
+```
+/**---------------------------------------------------------------------------*
+ * @brief Construct a new column from a size, type, and option to
+ * allocate bitmask.
+ * 
+ * Both the data and bitmask are unintialized.
+ *
+ * @param[in] size The number of elements in the column
+ * @param[in] type The element type
+ * @param[in] allocate_bitmask Optionally allocate an appropriate sized
+ * bitmask
+ *---------------------------------------------------------------------------**/
+column(int size, DType type, bool allocate_bitmask = false);
+
+/**---------------------------------------------------------------------------*
+ * @brief Construct a new column from a type, and device_buffers for
+ * data and bitmask that will be *deep* copied.
+ *
+ * @param[in] dtype The element type
+ * @param[in] data_buffer device_buffer whose data will be *deep* copied
+ * @param[in] mask_buffer Optional device_buffer whose data will be *deep*
+ *copied
+ *---------------------------------------------------------------------------**/
+column(DType dtype, device_buffer data,
```

Comment:
nit: why is data and mask_buffer const references on the host side and not on the device side?

Reply:
Related to https://github.com/jrhemstad/cudf_column/pull/1#discussion_r287502365

When accepting a `host_buffer`, we're only ever going to _deep_ copy the data to device memory, and we don't want to extraneously create a host copy by passing the `host_buffer` by value. Likewise, I don't expect we'll ever need to `move` a `host_buffer` into a `column` because the `column` doesn't own host data.

Make sense?